src/commands/revealResource.ts

```diff
@@ -3,26 +3,56 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { parseAzureResourceId } from '@microsoft/vscode-azext-azureutils';
+import { ParsedAzureResourceId } from '@microsoft/vscode-azext-azureutils';
 import { AzExtTreeItem, IActionContext, parseError } from '@microsoft/vscode-azext-utils';
-import { AppResource } from '@microsoft/vscode-azext-utils/hostapi';
+import { VSCodeRevealOptions } from '@microsoft/vscode-azext-utils/hostapi.v2';
 import { ext } from '../extensionVariables';
 import { ResourceGroupsItem } from '../tree/v2/ResourceGroupsItem';
 import { ResourceTreeDataProviderBase } from '../tree/v2/ResourceTreeDataProviderBase';
 
-export async function revealResource(context: IActionContext, resourceId: string): Promise<void>;
-export async function revealResource(context: IActionContext, resource: AppResource): Promise<void>;
-export async function revealResource(context: IActionContext, arg: AppResource | string): Promise<void> {
-    const resourceId = typeof arg === 'string' ? arg : arg.id;
-
-    context.telemetry.properties.resourceType = parseAzureResourceId(resourceId).provider.replace(/\//g, '|');
+export async function revealResource(context: IActionContext, resourceId: string, options?: VSCodeRevealOptions): Promise<void> {
+    setTelemetryPropertiesForId(context, resourceId);
 
     try {
         const item: ResourceGroupsItem | undefined = await (ext.v2.api.resources.azureResourceTreeDataProvider as ResourceTreeDataProviderBase).findItemById(resourceId);
         if (item) {
-            await ext.appResourceTreeView.reveal(item as unknown as AzExtTreeItem, { expand: false, focus: true, select: true });
+            await ext.appResourceTreeView.reveal(item as unknown as AzExtTreeItem, options ?? { expand: false, focus: true, select: true });
         }
     } catch (error) {
         context.telemetry.properties.revealError = parseError(error).message;
     }
 }
+
+function setTelemetryPropertiesForId(context: IActionContext, resourceId: string): void {
+    const parsedAzureResourceId = parsePartialAzureResourceId(resourceId);
+    const resourceKind = getResourceKindFromId(parsedAzureResourceId);
+    context.telemetry.properties.resourceKind = resourceKind;
+
+    if (resourceKind === 'resource') {
+        context.telemetry.properties.resourceType = parsedAzureResourceId.provider?.replace(/\//g, '|');
+    }
+}
+
+function parsePartialAzureResourceId(id: string): Partial<ParsedAzureResourceId> & Pick<ParsedAzureResourceId, 'rawId'> {
```

Comment:
I thought we had utility methods for this in `vscode-azuretools`.

Reply:
The utility we have expects a full ARM ID, and throws an error if you pass a partial id.